embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/anr/AnrOtelMapper.kt

```diff
@@ -0,0 +1,79 @@
+package io.embrace.android.embracesdk.anr
+
+import io.embrace.android.embracesdk.arch.DataCaptureServiceOtelConverter
+import io.embrace.android.embracesdk.internal.clock.millisToNanos
+import io.embrace.android.embracesdk.internal.payload.Attribute
+import io.embrace.android.embracesdk.internal.payload.Span
+import io.embrace.android.embracesdk.internal.payload.SpanEvent
+import io.embrace.android.embracesdk.payload.AnrInterval
+import io.embrace.android.embracesdk.payload.AnrSample
+import io.opentelemetry.sdk.trace.IdGenerator
+
+/**
+ * Maps captured ANRs to OTel constructs.
+ */
+internal class AnrOtelMapper(
+    private val anrService: AnrService
+) : DataCaptureServiceOtelConverter {
+
+    override fun snapshot(): List<Span> {
+        val intervals = anrService.getCapturedData()
+
+        return intervals.map { interval ->
+            val attrs = mapIntervalToSpanAttributes(interval)
+            val events = mapIntervalToSpanEvents(interval)
+            Span(
+                traceId = null,
```

Comment:
Can you use `SpanService` to create this rather than create the payload object directly? This will make it set the right defaults and make this a lot more resilient to change.